speech/snippets/context_classes.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package snippets contains speech examples.
+package snippets
+
+import (
+	speech "cloud.google.com/go/speech/apiv1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1"
+
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// [START speech_context_classes]
+
+// contextClasses provides "hints" to the speech recognizer
+// to favour specific classes of words in the results
+func contextClasses(w io.Writer, gcsURI string) error {
+	ctx := context.Background()
+
+	client, err := speech.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewClient: %v", err)
+	}
+
```

Comment:
Add a `defer client.Close()` after the error check block to release client resources after you're done with them.

Reply:
hmm should this be after the 2nd err != nil?

Reply:
Create client -> check err -> defer close. So, right where this comment is.